junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -385,16 +399,27 @@ private SortedMap<Path, IOException> deleteAllFilesAndDirectories(FileOperations
 			Files.walkFileTree(this.dir, new SimpleFileVisitor<Path>() {
 
 				@Override
-				public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
+				public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+					LOGGER.trace(() -> "preVisitDirectory: " + dir);
+					if (isLink(dir)) {
+						return SKIP_SUBTREE;
+					}
```

Comment:
:speech_balloon: This avoids following junctions and, potentially, deleting files outside of the temp dir.